x-pack/plugins/apm/server/lib/transaction_groups/transform.ts

```diff
@@ -27,7 +34,7 @@ function getTransactionGroup(
   const averageResponseTime = bucket.avg.value;
   const transactionsPerMinute = bucket.doc_count / minutes;
   const impact = bucket.sum.value;
-  const sample = bucket.sample.hits.hits[0]._source;
+  const sample = bucket.sample.hits.hits[0]._source as Transaction;
```

Comment:
That works for the top-level `hits` object, but this is from a `top_hits` aggregation, so it needs a type hint. I'm not sure how to solve it without the type hint because it's so dynamic. Do you have any ideas?

Reply:
in this case, i think you're right to use the type assertion